aten/src/TH/THTensor.hpp

```diff
@@ -63,6 +63,14 @@ struct THTensor
       return sizes_.size();
     }
 
+    at::ScalarType scalar_type() const {
+      return storage_->scalar_type;
+    }
+
+    at::Backend backend() const {
+      return storage_->data_ptr.device().backend();
```

Comment:
I'd prefer not to put backend in the public interface, because backend sucks.  Just make this an external function.